src/gui_common/CustomRichTextLabel.cs

```diff
@@ -225,32 +225,43 @@ private string BuildTemplateForTag(string input, ThriveBbCode bbcode, List<strin
         // Defaults to input so if something fails output returns unchanged
         var output = input;
 
+        var simulationParameters = SimulationParameters.Instance;
+
         switch (bbcode)
         {
             case ThriveBbCode.Compound:
             {
-                if (!SimulationParameters.Instance.DoesCompoundExist(input))
-                {
-                    GD.Print($"Compound: \"{input}\" doesn't exist, referenced in bbcode");
-                    break;
-                }
-
-                var compound = SimulationParameters.Instance.GetCompound(input);
-
-                var name = compound.Name;
-
                 var pairs = StringUtils.ParseKeyValuePairs(attributes);
 
-                // Parse attributes if there are any
-                if (pairs.TryGetValue("text", out string value))
+                var fallback = false;
+                var internalName = string.Empty;
+
+                if (pairs.TryGetValue("type", out string value))
                 {
                     if (!value.StartsAndEndsWith("\""))
                         break;
 
-                    name = value.Substring(1, value.Length - 2);
+                    internalName = value.Substring(1, value.Length - 2);
                 }
 
-                output = $"[b]{name}[/b] [font=res://src/gui_common/fonts/" +
+                if (!string.IsNullOrEmpty(internalName) && !simulationParameters.DoesCompoundExist(internalName))
+                {
+                    GD.Print($"Compound: \"{internalName}\" doesn't exist, referenced in bbcode");
+                    break;
+                }
+
+                // If type attribute is not specified but input is valid internal name, fallback to the old method
+                if (string.IsNullOrEmpty(internalName) && simulationParameters.DoesCompoundExist(input))
```

Comment:
`if (internalName == string.empty && ...`

Reply:
I think it could be left as is. `string.IsNullOrEmpty` is already used almost everywhere in the codebase and I don't see real benefit of changing it in favor of direct string comparison.